src/ext.js

```diff
@@ -97,230 +99,349 @@ const textAlignItems = {
+            libraryItemType: 'measure',
+            ref: 'qLibraryId',
+            translation: 'Common.Measure',
+            show: (itemData) => itemData.qLibraryId,
+          },
+          inlineMeasure: {
+            component: 'inline-measure',
+            show: (itemData) => !itemData.qLibraryId,
+          },
+          ...columnCommonHidden,
+          ...columnExpressionItems,
+          ...textAlignItems,
+          totalsAggr: {
+            type: 'items',
+            grouped: true,
+            items: {
+              totalsAggrGroup: {
+                type: 'items',
+                items: {
+                  totalsAggrFunc: {
+                    type: 'string',
+                    component: 'dropdown',
+                    ref: 'qDef.qAggrFunc',
+                    translation: 'Object.Table.AggrFunc',
+                    options(data, handler) {
+                      const hasActiveModifiers = Modifiers.hasActiveModifiers({
+                        measures: [data],
+                        properties: handler.properties,
+                      });
+                      const enableTotal = !hasActiveModifiers || Modifiers.ifEnableTotalsFunction(data);
+                      const autoOption = enableTotal
+                        ? [
+                            {
+                              value: 'Expr',
+                              translation: 'Common.Auto',
+                            },
+                          ]
+                        : [];
+                      return autoOption.concat([
+                        {
+                          value: 'Avg',
+                          translation: 'Object.Table.AggrFunc.Avg',
+                        },
+                        {
+                          value: 'Count',
+                          translation: 'Object.Table.AggrFunc.Count',
+                        },
+                        {
+                          value: 'Max',
+                          translation: 'Object.Table.AggrFunc.Max',
+                        },
+                        {
+                          value: 'Min',
+                          translation: 'Object.Table.AggrFunc.Min',
+                        },
+                        {
+                          value: 'Sum',
+                          translation: 'Object.Table.AggrFunc.Sum',
+                        },
+                        {
+                          value: 'None',
+                          translation: 'Object.Table.AggrFunc.None',
+                        },
+                      ]);
+                    },
+                    defaultValue: env?.anything?.sense?.isUnsupportedFeature?.('totals') ? 'None' : 'Expr',
+                  },
+                },
+              },
             },
+            show: !env?.anything?.sense?.isUnsupportedFeature?.('totals'),
           },
         },
       },
     },
-    settings: {
-      uses: 'settings',
+  };
```

Comment:
The presentation, total functions. They are added via the current PR, the other changes are mostly for refactoring purposes which was required.